src/Domain/ProductAttribute/Services/AttributeOptionService.php

```diff
@@ -84,4 +87,28 @@ public function importSku(Attribute $attribute, Product $product, string $locale
                 ?->product_attribute_pivot?->options()->sync([$option->getKey()]);
         }
     }
+
+    private function checkIfValueChanged(AttributeOption $option, AttributeOptionDto $dto, AttributeType $type): bool
+    {
+        return match ($type) {
+            AttributeType::NUMBER => $option->value_number !== $dto->value_number,
+            AttributeType::DATE => $option->value_date !== $dto->value_date,
+            AttributeType::SINGLE_OPTION, AttributeType::MULTI_CHOICE_OPTION => $this->checkIfNameChanged($option->name, $dto->translations),
+        };
+    }
+
+    /**
+     * @param array<string, array<string, string>>|null $translations
+     */
+    private function checkIfNameChanged(string $name, ?array $translations): bool
+    {
+        /** @var string $defaultLanguage */
+        $defaultLanguage = App::make(LanguageService::class)->defaultLanguage()->getKey();
+
+        if ($translations && array_key_exists($defaultLanguage, $translations)) {
```

Comment:
This `if` block could be replaced with
`return Arr::get($translations ?? [], $defaultLanguage . '.name', $name) !== $name`

If $translations array is empty or key does not exists, it will return default value of $name, so return value of this method will be false.